model/src/main/java/org/mskcc/cmo/metadb/model/MetadbRequest.java

```diff
@@ -381,10 +381,14 @@ public void updateRequestMetadataByMetadata(RequestMetadata requestMetadata)
         this.libraryType = String.valueOf(metadataMap.get("libraryType"));
         this.bicAnalysis = Boolean.parseBoolean(String.valueOf(metadataMap.get("bicAnalysis")));
         this.isCmoRequest = Boolean.parseBoolean(String.valueOf(metadataMap.get("isCmoRequest")));
-        this.metaDbProject = new MetadbProject(igoRequestId.split("_")[0]);
         addRequestMetadata(requestMetadata);
     }
 
+    public String resolveGenePanel(Map<String, Object> metadataMap) {
+        return String.valueOf(ObjectUtils.firstNonNull(metadataMap.get("recipe"),
```

Comment:
don't return the `String.valueOf(..)` here since it might be a null and we'd end up with a `"null"` value.

Instead look at what's returned by ObjectUtils... and then if null return an empty string